docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,139 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
+The constructor must be specified as a method descriptor and must only contain the parameters visible in the source code, omitting the hidden constant name and ordinal parameters.
 
-For normal, non-`StringRepresentable` enums, the codec can be wrapped via `Codec#lazyInitialized` or [`NeoForgeStreamCodecs#lazy`][streamcodec]. These prevent the codec from being resolved until first usage, which will always be after all mod entries are added.
+#### Parameters
 
-```java
-// For some enum with codec
-public enum ExampleEnumCodec {
-    // ...
-    ;
-
-    public static final Codec<ExampleEnumCodec> CODEC = Codec.of(/* ... */);
-    public static final StreamCodec<ByteBuf, ExampleEnumCodec> CODEC = StreamCodec.of(/* ... */);
+The parameters can be specified in three ways with limitations depending on the parameter types:
 
-    // ...
-}
+- Inline in the JSON file as an array of constants (only allowed for primitive values, Strings and for passing null to any reference type)
+- As a reference to a field of type `EnumProxy<TheEnum>` in a class from the mod (see `EnumProxy` example above)
+- As a reference to a method returning `Object`, where the return value is the parameter value to use. The method must have exactly two parameters of type `int` (index of the parameter) and `Class<?>` (expected type of the parameter)
+  - The `Class<?>` object should be used to cast (`Class#cast()`) the return value in order to keep `ClassCastException`s in mod code.
 
-// The final extended enum may look something like this
-public enum ExampleEnumCodec implements net.neoforged.neoforge.common.IExtensibleEnum {
-    // ...
-    ;
-
-    public static final Codec<ExampleEnumCodec> CODEC = Codec.lazyInitialized(
-        () -> Codec.of(/* ... */)
-    );
-    public static final StreamCodec<ByteBuf, ExampleEnumCodec> CODEC = 
-        net.neoforged.neoforge.common.util.NeoForgeStreamCodecs.lazy(
-            () -> StreamCodec.of(/* ... */)
-        );
+:::warning
+The fields and/or methods used as sources for parameter values should be in a separate class to avoid unintentionally loading mod classes too early.
+:::
 
-    // ...
-}
-```
+Certain parameters have additional rules:
 
-For `StringRepresentable` enums, `IExtensibleEnum` provides `createCodecForExtensibleEnum` and `createStreamCodecForExtensibleEnum` which do not cache the values at all, meaning they will always obtain up-to-date enums added at any point.
+- If the parameter is an int ID parameter related to a `@IndexedEnum` annotation on the enum, then it is ignored and replaced by the entry's ordinal. If said parameter is specified inline in the JSON, then it must be specified as `-1`.
```

Comment:
Does this throw an error if not -1? And how does this work if the index enum is not the first parameter (I don't think this happens in vanilla, but I'm unsure whether ignore means only if its the first parameter).